### PR TITLE
Avoid sending a string to `/traces`

### LINKED ISSUE #300: chore: change `dl_vlan` sent to trace to avoid error with strings.
Need to avoid sending strings to `sdn_trace` to ensure future changes.
Related `sdn_trace` [PR](https://github.com/kytos-ng/sdntrace_cp/pull/82)

### PR DESCRIPTION
**Pull request by gretelliz**
Closes #300 

### Summary

Map `dl_vlan` to an integer in range [1, 4095] for `/traces`.

### Local Tests

[See unit test](https://github.com/kytos-ng/mef_eline/blob/8b28b7796a9ccce525e69915f25f2fac3f78276a/tests/unit/models/test_evc_deploy.py#L1348-L1353)

### End-to-End Tests

N/A

### CODE REVIEW

File: tests/unit/models/test_evc_deploy.py [hunk 1]
```diff
@@ -1345,6 +1345,97 @@ def test_run_bulk_sdntraces(self, put_mock):
...
+                                    timeout=30
+                                )
+        args = put_mock.call_args[1]['json'][0]['trace']
+        assert args['eth'] == {'dl_type': 33024, 'dl_vlan': 1}
+
+        expected_payload[0]['trace']['eth'] = {
+            'dl_type': 0x8100,
+            'dl_vlan': 10
+            }
+        evc.uni_a.user_tag.value = '10/10'
```
Comment (viniarck) [inline tests/unit/models/test_evc_deploy.py:1417]: It's good to have this covered.

It'd be great if you could cover a less trivial mask example too, like "5/2" -> 0 and "5/3" -> 1.

**Comment by Alopalao** [comment]
Looks good. Just one thing, you can have and e2e test for this changed since it is from `mef_eline` side and not `flow_manager`. You can create a circuit for `"any"` and `"untagged"` and trace a path with `sdn_trace` api. wdyt @viniarck? or is it too many tests?

File: tests/unit/models/test_evc_deploy.py (hunk 1 above)
Comment (gretelliz) [reply tests/unit/models/test_evc_deploy.py:1417]: @viniarck, now I have a doubt. In the case "5/2 -> 0 and the similar, i.e. "20/10" --> "n1/n2" with [n1 & (n2 $4095)] == 0, the mapped value should be 0? We agreed that ` dl_vlan` would be in range [1,4095].

Comment (viniarck) [reply tests/unit/models/test_evc_deploy.py:1417]: Right, @gretelliz, `"5/2"` will result in 0, but since you have this if `if uni_dl_vlan`, then it'll be sent without a vlan header. This case here is a corner case, since we don't have validation for the mask length in the API yet and we're allowing any of the 12 bits to be 0s or 1s, then it can result in 0 (one day we might restrict that further too, for now we're keeping it very open since vlan range discussion is still going on). and `"5/3"` -> 1, which is why it's good to cover these cases with unit cases, since in the future if we decide to further restrict more then unit tests will catch it.

Comment (gretelliz) [reply tests/unit/models/test_evc_deploy.py:1417]: Great, everything is clear. Thank you.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -18,6 +18,7 @@ Changed
 - Hid ui primary and secondary constraints on ``k-toolbar`` in the meantime
 - Moved request circuit ``k-button`` out of k-accordion-item since it's mandatory
 - The traces being check rely on ``type``: ``last`` to be considered valid.
+- ``dl_vlan`` value is mapped to an integer in range [1, 4095] for the ``/traces``requests to ``sdntrace_cp``.
```
Comment (viniarck) [inline CHANGELOG.rst:21]: Looks like the extra backticks formatting as code ended up including more words than it should

**Comment by gretelliz** [comment]
I added e2e tests in `mef_eline` different from te one in [`sdntrace_cp`](https://github.com/kytos-ng/kytos-end-to-end-tests/pull/215). So, the direct call to `/traces` is in the e2e tests of `sdntrace_cp`. For `mef_eline` this endpoint is not used direct since what I want to test is that the special values of `dl_vlan` in the evcs are processed correctly. [See this PR](https://github.com/kytos/end-to-end-tests/pull/20), which verifies that there is a 'current_path' for the circuit created with the special `dl_vlan`.

**Comment by gretelliz** [comment]
Sure, I can

**Comment by viniarck** [comment]
> Sure, I can

@gretelliz, I was quickly exploring some cases here too, and I ended up hitting another related issue, I was using this any/any EVC:

```
{
  "name": "evpl3",
  "dynamic_backup_path": true,
  "uni_a": {
    "interface_id": "00:00:00:00:00:00:00:01:1",
    "tag": {
      "tag_type": 1,
      "value": "any"
    }
  },
  "uni_z": {
    "interface_id": "00:00:00:00:00:00:00:03:1",
    "tag": {
      "tag_type": 1,
      "value": "any"
    }
  }
}
```

The `compare_uni_out_trace` func will need to be updated, for instance, after a restart, notice that mef_eline's consistency check couldn't consider the circuit as active when checking the traces:

```
kytos $> 2023-03-30 14:25:16,467 - WARNING [kytos.napps.kytos/mef_eline] [evc.py:1216:check_trace] (mef_eline) Invalid trace from uni_a: [{'dpid': '00:00:00:00:00:00:00:01', 'port': 1, 
'time': '2023-03-30 14:25:16.466641', 'type': 'starting', 'vlan': 1}, {'dpid': '00:00:00:00:00:00:00:03', 'out': {'port': 1, 'vlan': 1}, 'port': 3, 'time': '2023-03-30 14:25:16.466683',
 'type': 'last', 'vlan': 6}]
2023-03-30 14:26:16,481 - WARNING [kytos.napps.kytos/mef_eline] [evc.py:1216:check_trace] (mef_eline) Invalid trace from uni_a: [{'dpid': '00:00:00:00:00:00:00:01', 'port': 1, 'time': '
2023-03-30 14:26:16.479935', 'type': 'starting', 'vlan': 1}, {'dpid': '00:00:00:00:00:00:00:03', 'out': {'port': 1, 'vlan': 1}, 'port': 3, 'time': '2023-03-30 14:26:16.479990', 'type': 
'last', 'vlan': 6}]
2023-03-30 14:27:16,533 - WARNING [kytos.napps.kytos/mef_eline] [evc.py:1216:check_trace] (mef_eline) Invalid trace from uni_a: [{'dpid': '00:00:00:00:00:00:00:01', 'port': 1, 'time': '
2023-03-30 14:27:16.531060', 'type': 'starting', 'vlan': 1}, {'dpid': '00:00:00:00:00:00:00:03', 'out': {'port': 1, 'vlan': 1}, 'port': 3, 'time': '2023-03-30 14:27:16.531140', 'type': 
'last', 'vlan': 6}]
```

With that said, can you fix this? In your local tests and or e2e would maybe will be great to also cover a restart there with any/any and untagged/untagged.

**Comment by viniarck** [comment]
> @viniarck , I just saw it, I was working on what you suggested before and I noticed this.

Cool : ) 

Glad you also caught it.

**Comment by gretelliz** [comment]
Fixed. Noticed that 2 new unit tests have been added to take this into account. [see](https://github.com/kytos-ng/mef_eline/blob/bdd9bedd486fb27f7af52f485d402daa0accd821/tests/unit/models/test_evc_deploy.py#L1575-L1703)

**Review by viniarck — APPROVED** [review]
Nicely done @gretelliz. Great how `map_dl_vlan(value)` ended up being reused. 

Although e2e tests are still being amended, this PR and `sdntrace_cp` has been heavily locally explored https://github.com/kytos-ng/sdntrace_cp/pull/82. If you wish you can merge these two, and then e2e can land shortly after since it's just minor refactoring details there.
